OpenSim/Simulation/Model/ModelComponent.cpp

```diff
@@ -124,8 +123,11 @@ void ModelComponent::addGeometry(OpenSim::Geometry& geom) {
         }
         
     }
-    append_geometry(geom);
-    return;
+
+    int ix = append_geometry(geom);
+    finalizeFromProperties();
```

Comment:
I think it's OK to require whoever called `addGeometry()` to have to finalize their model again. What if they were adding tons of geometry and thus only need to finalize once after they've added everything?